app/Http/Controllers/PostsController.php

```diff
@@ -143,10 +145,15 @@ public function update(Request $request, $id)
             $fileNameToStore= $filename.'_'.time().'.'.$extension;
             // Upload Image
             $path = $request->file('cover_image')->storeAs('public/cover_images', $fileNameToStore);
+            
+            // Delete Previous Image
+            if($post->cover_image != 'noimage.jpg') {
+                Storage::delete('public/cover_images/'.$post->cover_image);
```

Comment:
maybe need to check if file on directory is really exist, then delete if has one.